tmail-backend/webadmin/webadmin-email-address-contact/src/main/java/com/linagora/tmail/webadmin/EmailAddressContactRoutes.java

```diff
@@ -187,6 +188,27 @@ public Route updateContact() {
         });
     }
 
+    private Mono<ContactFields> fillUpMissingNameFields(Domain domain, ContactFields updatedFields) {
+        if (updatedFields.firstname().isEmpty() || updatedFields.surname().isEmpty()) {
+            return Mono.from(emailAddressContactSearchEngine.get(domain, updatedFields.address()))
+                .map(contact -> fillUp(contact.fields(), updatedFields))
+                .defaultIfEmpty(updatedFields);
+        }
+        return Mono.just(updatedFields);
+    }
+
+    private ContactFields fillUp(ContactFields presentFields, ContactFields updatedFields) {
+        String firstname = updatedFields.firstname();
+        String surname = updatedFields.surname();
+        if (firstname.isEmpty()) {
+            firstname = presentFields.firstname();
+        }
+        if (surname.isEmpty()) {
+            surname = presentFields.surname();
+        }
```

Comment:
Sorry to bother you again. I think we should not update names when the field is null but should update when it is empty.
E.g: null cases
```
{
  "firstname": "a firstname"
}
```
or 
```
{
	"firstname": "a firstname",
	"surname": null
}
```
IMO if we use Optional we can easily parse this. Optional.empty would be diff from an empty string like this (which produce Optional.of(""):
```
{
	"firstname": "a firstname",
	"surname": ""
}
```

Reply:
Good point, I omitted that case. Will solve it monday